tfe/resource_tfe_oauth_client.go

```diff
@@ -79,22 +108,37 @@ func resourceTFEOAuthClientCreate(d *schema.ResourceData, meta interface{}) erro
 
 	// Get the organization and provider.
 	organization := d.Get("organization").(string)
+	name := d.Get("name").(string)
 	privateKey := d.Get("private_key").(string)
+	rsaPublicKey := d.Get("rsa_public_key").(string)
+	key := d.Get("key").(string)
+	secret := d.Get("secret").(string)
 	serviceProvider := tfe.ServiceProviderType(d.Get("service_provider").(string))
 
 	if serviceProvider == tfe.ServiceProviderAzureDevOpsServer && privateKey == "" {
 		return fmt.Errorf("private_key is required for service_provider %s", serviceProvider)
 	}
 
 	// Create a new options struct.
+	// The tfe.OAuthClientCreateOptions has omitempty for these values, so if it
+	// is empty, then it will be ignored in the create request
 	options := tfe.OAuthClientCreateOptions{
+		Name:            tfe.String(name),
 		APIURL:          tfe.String(d.Get("api_url").(string)),
 		HTTPURL:         tfe.String(d.Get("http_url").(string)),
 		OAuthToken:      tfe.String(d.Get("oauth_token").(string)),
-		PrivateKey:      tfe.String(privateKey),
+		Key:             tfe.String(key),
 		ServiceProvider: tfe.ServiceProvider(serviceProvider),
 	}
 
+	if serviceProvider == tfe.ServiceProviderAzureDevOpsServer {
+		options.PrivateKey = tfe.String(privateKey)
```

Comment:
I moved this here because `private_key` is only related to ADO. See the error in line 118.